### ISSUE TITLE
TASK [galaxy/com.devture.ansible.role.traefik : Ensure provider.yml installed] *** FAILED

### ISSUE THREAD

**Issue by UehgxEQbFiw4Tlw8** [spantaleev/matrix-docker-ansible-deploy#2844]
**Describe the bug**
Hello everyone!

First of all thank you so much to all of you for your efforts in this project! Extremely helpful for people like me (not very skilled when it comes to IT issues etc.).

My problem: After git pull and running the playbook as described in the documentation, suddenly:

TASK [galaxy/com.devture.ansible.role.traefik : Ensure provider.yml installed] ***
fatal: [matrix.exemple.com]: FAILED! => 
  msg: |-
    The task includes an option with an undefined variable. The error was: {{ devture_traefik_provider_configuration_yaml | from_yaml | combine(devture_traefik_provider_configuration_extension, recursive=True) }}: {{ devture_traefik_provider_configuration_extension_yaml | from_yaml if devture_traefik_provider_configuration_extension_yaml | from_yaml is mapping else {} }}: http:
     routers:
       {% for host in groups['jitsi_jvb_servers'] %}
  
       additional-{{ hostvars[host]['jitsi_jvb_server_id'] }}-router:
         entryPoints:
           - "{{ devture_traefik_entrypoint_primary }}"
         rule: "Host(`{{ jitsi_hostname }}`) && PathPrefix(`/colibri-ws/{{ hostvars[host]['jitsi_jvb_server_id'] }}/`)"
         service: additional-{{ hostvars[host]['jitsi_jvb_server_id'] }}-service
         {% if devture_traefik_entrypoint_primary != 'web' %}
  
         tls:
           certResolver: "{{ devture_traefik_certResolver_primary }}"
  
         {% endif %}
  
       {% endfor %}
  
     services:
       {% for host in groups['jitsi_jvb_servers'] %}
  
       additional-{{ hostvars[host]['jitsi_jvb_server_id'] }}-service:
         loadBalancer:
           servers:
             - url: "http://{{ host }}:9090/"
  
       {% endfor %}
    : 'dict object' has no attribute 'jitsi_jvb_servers'
  
    The error appears to be in '/home/myname/matrix-docker-ansible-deploy/roles/galaxy/com.devture.ansible.role.traefik/tasks/install.yml': line 73, column 3, but may
    be elsewhere in the file depending on the exact syntax problem.
  
    The offending line appears to be:
  
  
    - name: Ensure provider.yml installed
      ^ here

PLAY RECAP *********************************************************************
matrix.exemple.com            : ok=316  changed=6    unreachable=0    failed=1    skipped=530  rescued=0    ignored=0   

**To Reproduce**
My `vars.yml` file looks like this: very minimal instance, few users,  no federation.

matrix_domain: example.com
matrix_homeserver_implementation: synapse
matrix_homeserver_generic_secret_key: 'REDACTED'
matrix_ssl_lets_encrypt_support_email: me@mymail.tld
devture_postgres_connection_password: 'REDACTED'
matrix_client_element_enabled: false
matrix_synapse_federation_enabled: false
matrix_coturn_enabled: true
matrix_architecture: "arm64"
matrix_dynamic_dns_enabled: true
matrix_dynamic_dns_domain_configurations:
  - provider: REDACTED.com
    protocol: dyndns2
    username: username
    password: REDACTED
    domain: XXXX.tld

matrix_playbook_reverse_proxy_type: playbook-managed-traefik
devture_traefik_config_certificatesResolvers_acme_email: me@mymail.tld

```yaml
Paste your vars.yml file here.
Make sure to remove any secret values before posting your vars.yml file publicly.
```

nothing special, just the above, I can send and receive messages. Of course - see above - no jitsi used with the system.

**Expected behavior**

No such fatal ...FAILED error message

**Matrix Server:**
 - OS: [e.g. Ubuntu 21.04] last raspiOS
 - Architecture [e.g. amd64, arm32, arm64] arm32 (armv7)

**Ansible:**
If your problem appears to be with Ansible, tell us:
- where you run Ansible -- e.g. on the Matrix server itself; on another computer (which OS? distro? standard installation or containerized Ansible?) on a laptop, not on the server itself; Ubuntu 23.04 LTS; 

- what version of Ansible you're running (see `ansible --version`)
ansible [core 2.12.8]

**Client:**

**Additional context**
As I said, Matrix / Synapse seems to work, just the error appears - I can live with it, but it shows an unsolved problem with the software. Maybe correlated to the last changes about Jitsi during the last few hours?

Thank you!

Best regards, 
M.

**Comment by spantaleev** [comment]
Seems to be caused by https://github.com/spantaleev/matrix-docker-ansible-deploy/pull/2842

Unless your indentation of `devture_traefik_provider_configuration_extension_yaml` is incorrect or something.

What does your `inventory/hosts` file look like? Is there a `jitsi_jvb_servers` group there?

**Comment by UehgxEQbFiw4Tlw8** [comment]
Hello,

thanks for the quick reply!

My inventory/hosts, just one line:

[matrix_servers]
matrix.example.com ansible_host=AAA.BBB.CCC.DDD. ansible_ssh_user=pi become=true become_user=root

It has happened after a fresh git pull
- where may I find the pace in order to verify the supposed indentation problem (if this can help you?)

Best
M.

**Comment by natoleet** [comment]
I was bitten by the same issue under the same circumstances. Started happening after a git pull moments ago.

all things in my var.yml file pertaining to traefik are as follows;
```
devture_traefik_config_entrypoint_web_secure_enabled: false
devture_traefik_config_entrypoint_web_forwardedHeaders_insecure: true
devture_traefik_container_web_host_bind_port: '0.0.0.0:81'
devture_traefik_additional_entrypoints_auto:
  - name: matrix-federation
    port: 8448
    host_bind_port: '0.0.0.0:8449'
    config: {}
```

I too only have a single line in my hosts file
```
[matrix_servers]
matrix.example.com ansible_host=10.x.x.x ansible_connection=local ansible_ssh_user=user become=true become_user=root ansible_python_interpreter=/usr/bin/python3
```

Also, I do not have jitsi enabled as I don't use it.

**Comment by spantaleev** [comment]
Fixed by a036987ba45d723f38ab8415868d65f276ee7336 and 4873b1800086fc766f579a2cd80d6372832eb316

People who use Jitsi with additional JVBs and Traefik will now need to define `devture_traefik_provider_configuration_extension_yaml` manually, as documented in `docs/configuring-playbook-jitsi.md`
